numpy/core/include/numpy/npy_math.h

```diff
@@ -210,8 +210,10 @@ double npy_spacing(double x);
         (sizeof (x) == sizeof (long double) ? _npy_signbit_ld (x) \
          : sizeof (x) == sizeof (double) ? _npy_signbit_d (x) \
          : _npy_signbit_f (x))
+    #define true_npy_signbit(x) (signbit((x)) != 0 ? 1 : 0)
 #else
     #define npy_signbit(x) signbit((x))
+    #define true_npy_signbit(x) (signbit((x)) != 0 ? 1 : 0)
```

Comment:
as this is a public header this would need to be named npy_true_signbit
I also wonder if we really need this as a public macro, shouldn't harm but it would need documentation